java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -411,6 +411,341 @@ public boolean isClosed() {
+
+        Stream<Map<String, Object>> backingStream =
+                StreamSupport.stream(Spliterators.spliteratorUnknownSize(
+                        itr, additionalCharacteristics), false);
+
+        Stream<MantaObject> stream = backingStream
+                .map(MantaObjectConversionFunction.INSTANCE)
+                .onClose(itr::close);
+
+        danglingStreams.add(stream);
+
+        return stream;
+    }
+
+
+    /**
+     * Moves a bucket from one path to another path. This operation is not
+     * transactional and may fail or produce inconsistent result if the source
+     * or the destination is modified while the operation is in progress.
+     *
+     * @param source Original path to move from
+     * @param destination Destination path to move to
+     * @param entry Bucket supplemental data object
+     * @throws IOException thrown when something goes wrong
+     * @throws MantaClientHttpResponseException Source doesn't exist or destination has invalid name
+     * @throws MantaClientException If this operation is applied to a directory.
+     */
+    private void moveBucket(final String source, final String destination,
+                            final MantaObjectResponse entry)
+            throws IOException {
+        try {
+            createBucket(destination);
+        } catch (MantaClientHttpResponseException e) {
+            if (e.getServerCode().equals(MantaErrorCode.INVALID_BUCKET_NAME_ERROR)) {
+                final String msg = String.format("Invalid Bucket Name used for destination: %s", destination);
+                throw new MantaClientException(msg, e.getCause());
+            }
+        }
+
+        MantaHttpHeaders sourceHeaders = entry.getHttpHeaders();
+        Long contentsCount = sourceHeaders.getResultSetSize();
+
+        /* If we were just copying an empty bucket, we just create the
+           new bucket and delete the original */
+        if (contentsCount != null && contentsCount == 0L) {
+            deleteBucket(source);
+            return;
+        }
+
+        MantaObjectResponse destBucket = head(destination);
+        String destBucketPath = destBucket.getPath();
+        String sourceBucketPath = entry.getPath();
+
+        listBucketObjects(source).forEach(mantaObject -> {
+            try {
+                String sourcePath = mantaObject.getPath();
+                String relPath = sourcePath.substring(sourceBucketPath.length());
+                String destFullPath = destBucketPath + SEPARATOR + relPath;
+
+                if (mantaObject.isBucketObject()) {
+                    move(mantaObject.getPath(), destFullPath);
+                }
+                if (mantaObject.isDirectory()) {
+                    final String msg = String.format("moveBucket operation applied in a directory: %s ", sourcePath);
+                    throw new MantaClientException(msg);
+                }
+            } catch (IOException e) {
```

Comment:
Please do not do this if you method signature supports `IOException`.

Reply:
`moveBucket` removed as a result of design discussions convened on buckets last week.